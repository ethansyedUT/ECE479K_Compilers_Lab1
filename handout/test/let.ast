#8
_program
  #8
  _class
    Main
    Object
    "let.cl"
    (
    #7
    _method
      main
      Int
      #7
      _block
        #6
        _let
          x
          Int
          #3
          _no_expr
          : _no_type
          #6
          _let
            y
            Bool
            #3
            _bool
              1
            : Bool
            #6
            _let
              x
              Int
              #4
              _int
                2
              : Int
              #5
              _int
                2
              : Int
            : Int
          : Int
        : Int
      : Int
    )
