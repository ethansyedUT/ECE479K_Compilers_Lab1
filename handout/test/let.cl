class Main {
    main() : Int {{
        (let x : Int <- 2, y : Bool in
           let x : Int in
            2
        );
    }};
};