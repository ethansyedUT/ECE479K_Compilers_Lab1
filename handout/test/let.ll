declare i32 @strcmp(i8*, i8*)
declare i32 @printf(i8*, ...)
declare void @abort()
declare i8* @malloc(i32)
define i32 @Main_main() {

main:
	%vtpm.0 = alloca i32
	store i32 0, i32* %vtpm.0
	%vtpm.1 = alloca i1
	store i1 true, i1* %vtpm.1
	%vtpm.2 = alloca i32
	store i32 2, i32* %vtpm.2
	ret i32 2
}

@.str = internal constant [25 x i8] c"Main_main() returned %d\0A\00"
define i32 @main() {

entry:
	%vtpm.3 = call i32 @Main_main(  )
	%vtpm.4 = getelementptr [25 x i8], [25 x i8]* @.str, i32 0, i32 0
	%vtpm.5 = call i32(i8*, ... ) @printf( i8* %vtpm.4, i32 %vtpm.3 )
	ret i32 0
}

