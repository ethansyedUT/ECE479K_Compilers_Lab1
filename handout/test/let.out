Main_main() returned 2
